hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/AuthenticationBaseMessageTask.java

```diff
@@ -115,8 +116,7 @@ private AuthenticationStatus authenticate() {
         } else if (clientEngine.getSecurityContext() != null) {
             return authenticate(clientEngine.getSecurityContext());
         } else if (credentials instanceof UsernamePasswordCredentials) {
-            UsernamePasswordCredentials usernamePasswordCredentials = (UsernamePasswordCredentials) credentials;
-            return verifyClusterName(usernamePasswordCredentials);
+            return verifyClusterName();
```

Comment:
how come we are not using the username and password here?

Reply:
The cluster name (to-be-compared) is added as a new field in the client protocol authentication messages (see this PR: hazelcast/hazelcast-client-protocol#248).